transport/rif_comms/node.py

```diff
@@ -83,36 +83,17 @@ def _receive_messages(self):
         Iterate over the Notification stream and block thread to receive messages.
         """
         for notification in self._our_topic_stream:
-            raiden_message = self._notification_to_message(notification.channelNewData)
-            if raiden_message:
-                self.log.info("incoming message", message=raiden_message)
-                self._handle_message(raiden_message)
+            payload = notification_to_payload(notification)
 
-    @staticmethod
-    def _notification_to_message(notification_data: ChannelNewData) -> RaidenMessage:
-        """
-        :param notification_data: raw data received by the RIF Comms GRPC API
-        :return: a raiden.Message
-        """
-        content_text = notification_data.data
-        """
-        ChannelNewData has the following structure:
-            from: "16Uiu2HAm8wq7GpkmTDqBxb4eKGfa2Yos79DabTgSXXF4PcHaDhWJ"
-            data: "{\"type\":\"Buffer\",\"data\":[104,101,121]}"
-            nonce: "\216f\225\232d\023e{"
-            channel {
-              channelId: "16Uiu2HAm9otWzXBcFm7WC2Qufp2h1mpRxK1oox289omHTcKgrpRA"
-            }
-        """
-        if content_text:
-            # we first transform the content of the notification data to a dictionary
-            content = json.loads(content_text.decode())
-            # the message is inside the notification data, encoded by the RIF Comms GRPC api
-            message_string = bytes(content["data"]).decode()
-            print("Message received", message_string)
-            message_dict = json.loads(message_string)
-            return message_from_dict(message_dict)
-        return None
+            try:
+                message_dict = json.loads(payload)
+                raiden_message = message_from_dict(message_dict)
+
+                if raiden_message:
+                    self.log.info("incoming message", message=raiden_message)
+                    self._handle_message(raiden_message)
+            except (TypeError, InvalidProtocolMessage):
+                self.log.error("incoming message could not be processed", payload=payload)
```

Comment:
once the payload is extracted, all this code is specific to processing raiden messages in a rif comms transport node